pystac_client/client.py

```diff
@@ -443,21 +443,26 @@ def get_collections(self) -> Iterator[Collection]:
                 call_modifier(self.modifier, collection)
                 yield collection
 
-    def get_items(
-        self, *ids: str, recursive: bool | None = None
-    ) -> Iterator["Item_Type"]:
+    def get_items(self, *ids: str, recursive: bool = False) -> Iterator["Item_Type"]:
```

Comment:
I'm not convinced we should change the function signature. If we need to change the underlying behavior, that might be ok.

Reply:
It's up to you but this overrides a method from `pystac.Catalog` and changes the function signature of the method that it overrides. In my experience, it is best not to change the interface of an inherited method unless absolutely necessary.

An example of how this sort of thing causes problems can be found here: https://github.com/stac-utils/pystac-client/pull/799#issuecomment-2898499159